modin/engines/base/frame/data.py

```diff
@@ -1672,12 +1674,14 @@ def _copartition(self, axis, other, how, sort, force_repartition=False):
 
         is_aligning_applied = False
         for i in range(len(other)):
+            # aligning partitions
             if (
                 len(self._partitions) != len(other[i]._partitions)
                 and len(self.axes[0]) == len(other[i].axes[0])
                 and axis == 0
             ):
                 is_aligning_applied = True
+                # we shouldn't modificate modin frame' partitions directly
                 self._partitions = self._frame_mgr_cls.map_axis_partitions(
                     axis, self._partitions, lambda df: df
                 )
```

Comment:
I know that these changes do not belong to this PR, but just pointing out that we're aligning `self` partitions in a loop, so we're aligning `self` partition `len(other) times`, when it's enough to make this only once, that logic definitely should be changed in the future, because we're getting here significant performance bottleneck

Reply:
Open issue https://github.com/modin-project/modin/issues/2374 for this.